src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/BLEUEvaluator.cs

```diff
@@ -0,0 +1,117 @@
+/// <see cref="BLEUEvaluatorContext.References"/>. The score is returned in a <see cref="NumericMetric"/>
+/// with a value between 0.0 and 1.0 where 0.0 represents no match at all and 1.0 indicates a perfect match.
+/// By default, the score is interpreted with a pass/fail cutoff of 0.5. So a score of 0.5 or higher is
+/// passing and a score below 0.5 is failing.
+/// </para>
+/// </remarks>
+[Experimental("AIEVAL001")]
+public sealed class BLEUEvaluator : IEvaluator
+{
+    /// <summary>
+    /// Gets the <see cref="EvaluationMetric.Name"/> of the <see cref="NumericMetric"/> returned by
+    /// <see cref="BLEUEvaluator"/>.
+    /// </summary>
+    public static string BLEUMetricName => "BLEU";
+
+    /// <inheritdoc/>
+    public IReadOnlyCollection<string> EvaluationMetricNames { get; } = [BLEUMetricName];
+
+    /// <inheritdoc/>
+    public ValueTask<EvaluationResult> EvaluateAsync(
+        IEnumerable<ChatMessage> messages,
+        ChatResponse modelResponse,
+        ChatConfiguration? chatConfiguration = null,
+        IEnumerable<EvaluationContext>? additionalContext = null,
+        CancellationToken cancellationToken = default)
+    {
+        Stopwatch stopwatch = Stopwatch.StartNew();
+
+        _ = Throw.IfNull(modelResponse);
+
+        var metric = new NumericMetric(BLEUMetricName);
+        var result = new EvaluationResult(metric);
+
+        if (string.IsNullOrWhiteSpace(modelResponse.Text))
+        {
+            metric.AddDiagnostics(
+                EvaluationDiagnostic.Error($"The {nameof(modelResponse)} supplied for evaluation was null or empty."));
+
+            return new ValueTask<EvaluationResult>(result);
+        }
+
+        if (additionalContext?.OfType<BLEUEvaluatorContext>().FirstOrDefault()
+                is not BLEUEvaluatorContext context)
+        {
+            metric.AddDiagnostics(
+                EvaluationDiagnostic.Error(
+                    $"A value of type '{nameof(BLEUEvaluatorContext)}' was not found in the '{nameof(additionalContext)}' collection."));
+
+            return new ValueTask<EvaluationResult>(result);
+        }
+
+        var references = context.References.Select(reference => SimpleWordTokenizer.WordTokenize(reference));
+        var hypothesis = SimpleWordTokenizer.WordTokenize(modelResponse.Text);
+        metric.Value = BLEUAlgorithm.SentenceBLEU(references, hypothesis, BLEUAlgorithm.DefaultBLEUWeights, SmoothingFunction.Method4);
+
+        stopwatch.Stop();
+        string durationText = $"{stopwatch.Elapsed.TotalSeconds.ToString("F2", CultureInfo.InvariantCulture)} s";
+
+        metric.AddOrUpdateContext(context);
+        metric.AddOrUpdateMetadata(name: "evaluation-duration", value: durationText);
+        metric.Interpretation = InterpretScore(metric);
+
+        return new ValueTask<EvaluationResult>(result);
+    }
+
+    private static EvaluationMetricInterpretation InterpretScore(NumericMetric metric)
```

Comment:
Consider: Assuming we'll need the same default interpretation across multiple evaluators, pull this helper out as an extension method in a new `EvaluationMetricExtensions` class like we do in the other packages.

https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.AI.Evaluation.Quality/EvaluationMetricExtensions.cs